entity_test.go

```diff
@@ -162,6 +163,16 @@ func TestRead_tooBig(t *testing.T) {
 	}
 }
 
+func TestReadWithOptions_canSetMaxHeaderBytes(t *testing.T) {
```

Comment:
You aren't checking any of the default value patching is working, only the manual override.
At the very least you should check for -1, 0 and some manual value in a test and validate that the option struct contains what you think it does.

Reply:
The existing Read func will check that 0 (int default) is mapped to something non 0 (otherwise it would fail). You're right I did not check -1, I'll add some more tests this evening.

Reply:
Now done.